source/MRViewer/MRRibbonNotification.cpp

```diff
@@ -32,18 +32,108 @@ namespace MR
+        notificationsPosX = float( menu->getSceneSize().x );
+    Vector2f currentPos = Vector2f( notificationsPosX, float( getViewerInstance().framebufferSize.y ) - 25.f * scaling );
+    const Vector2f padding = Vector2f( 0.0f, 20.0f * scaling );
+
+    
+    auto iconsFont = RibbonFontManager::getFontByTypeStatic( RibbonFontManager::FontType::Icons );
+    if ( iconsFont )
     {
-        if ( notifications_.size() == cNotificationNumberLimit )
-            notifications_.erase( notifications_.end() - 1 );
-        notifications_.insert( notifications_.begin(), NotificationWithTimer{ notification } );
+        iconsFont->Scale = 0.7f;
+        ImGui::PushFont( iconsFont );
     }
-    requestClosestRedraw_();
+    const float size = ImGui::GetTextLineHeight() + 10.f * scaling * 2.f;
+    if ( iconsFont )
+    {
+        iconsFont->Scale = 1.0f;
+        ImGui::PopFont();
+    }
+    
+    currentPos.x += padding.y;
+
+    currentPos -= padding;
+
+    ImGui::SetNextWindowPos( currentPos, ImGuiCond_Always, ImVec2( 0.f, 1.0f ) );
+    ImGui::SetNextWindowSize( ImVec2( size, size ), ImGuiCond_Always );
+    ImGuiWindowFlags flags =
+        ImGuiWindowFlags_AlwaysAutoResize |
+        ImGuiWindowFlags_NoResize |
+        ImGuiWindowFlags_NoTitleBar |
+        ImGuiWindowFlags_NoMove;
+    std::string name = "##NotificationButton";
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowBorderSize, 0.0f );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowRounding, 4.0f * scaling );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, ImVec2( 10.0f * scaling, 10.0f * scaling ) );
+    ImGui::PushStyleVar( ImGuiStyleVar_CellPadding, { 0, 0 } );
+    ImGui::PushStyleColor( ImGuiCol_WindowBg, MR::ColorTheme::getRibbonColor( MR::ColorTheme::RibbonColorsType::FrameBackground ).getUInt32() );
+
+    ImGui::Begin( name.c_str(), nullptr, flags );
+
+    iconsFont = RibbonFontManager::getFontByTypeStatic( RibbonFontManager::FontType::Icons );
+    if ( iconsFont )
+    {
+        iconsFont->Scale = 0.7f;
+        ImGui::PushFont( iconsFont );
+    }
+
+    ImGui::PushStyleColor( ImGuiCol_Text, notificationParams[int( highestNotification_ )].second );
+    ImGui::Text( "%s", notificationParams[int( highestNotification_ )].first );
+    ImGui::PopStyleColor();
+
+    if ( iconsFont )
+    {
+        iconsFont->Scale = 1.0f;
+        ImGui::PopFont();
+    }
+
+    auto window = ImGui::GetCurrentContext()->CurrentWindow;
+    bool isHovered = false;
+    isHovered = ImGui::IsWindowHovered();
+
+    if ( isHovered )
+    {
+        if ( ImGui::IsMouseClicked( ImGuiMouseButton_Left ) )
+        {
+            for ( int i = int( notificationsHistory_.size() ) - 1; i >= 0; --i )
+            {
+                const auto& nwt = notificationsHistory_[i];
+                for ( int j = 0; j < nwt.sameCounter; ++j )
+                    addNotification_( notifications_, nwt.notification );
+            }
+            requestClosestRedraw_();
+        }
```

Comment:
what if click several times? will `notifications_` timer work and close them automatical?